### PR TITLE
Loss functions

### LINKED ISSUE #96: More meaningful names in the generated json

### PR DESCRIPTION
**Pull request by ndslusarz**
Has work for issues #96 (changing the names to more meaningful - finished) and #97.

 #97 is partially solved - there is code handling Annotations, Lam and Let which we don't want normalised and checks type of function making sure it goes to Bool/Prop (but it's still missing the new normalisation to replace the full normalisation we were using before).

### CODE REVIEW

File: src/hs/Vehicle/Backend/LossFunction/Compile.hs [hunk 1]
```diff
@@ -35,17 +36,18 @@ instance ToJSON Domain where
...
 data LExpr
-  = Neg LExpr
-  | Con Double
+  = Negation LExpr
+  | Constant Double
   | Min LExpr LExpr
   | Max LExpr LExpr
-  | Sub LExpr LExpr
-  | Ind LExpr LExpr
+  | Substraction LExpr LExpr
```
Comment (MatthewDaggitt) [inline src/hs/Vehicle/Backend/LossFunction/Compile.hs]: Subtraction?

File: src/hs/Vehicle/Backend/LossFunction/Compile.hs [hunk 2]
```diff
@@ -35,17 +36,18 @@ instance ToJSON Domain where
...
+  | IndicatorFunction LExpr LExpr
   | Var V.DBIndex
```
Comment (MatthewDaggitt) [inline src/hs/Vehicle/Backend/LossFunction/Compile.hs]: As we're being verbose, can we have `Variable`, `NetworkApplication`, `Quantifier` and `TensorLiteral` as well?

**Review by MatthewDaggitt — CHANGES_REQUESTED** [review]
Otherwise looks good!

File: src/hs/Vehicle/Backend/LossFunction/Compile.hs (hunk 1 above)
Comment (ndslusarz) [reply src/hs/Vehicle/Backend/LossFunction/Compile.hs]: Corrected now :)

File: src/hs/Vehicle/Backend/LossFunction/Compile.hs (hunk 2 above)
Comment (ndslusarz) [reply src/hs/Vehicle/Backend/LossFunction/Compile.hs]: Done in the new commit.

**Review by MatthewDaggitt — APPROVED** [review]

**Comment by MatthewDaggitt** [comment]
Fixed the merge conflicts, but when I run the tests via `cabal run build test`, I get an error saying there's still no case for translating variables! We'd probably best add one!

**Pull request merged**